### PR TITLE
ypang week02 submission

### PR DESCRIPTION
**Pull request by andrewytp**
This week we learnt how to plot graphs in python using matlibplot and scipy. We also learnt to avoid overfitting by using BIC

In the assignment, 
linear fit: BIC =  29.979491346722153
cubic fit: BIC =  30.360576141400536

Since linear fit having lower BIC, it is a better choice of fitting.

### CODE REVIEW

**Comment by andrewytp** [comment]
I have execute the commands that you suggest, and resolved conflicts. Though I am not sure the cause and how to avoid it next time. Is that because I did not create a new branch before I move on to week 2 assignment? (but I though I did. I just delete the branch when you accepted my week01 pull request) Or is that because I used pull instead of fetch + merge on the upstream/master? (I though they are the same?) Or is that because I have to wait until you confirm my previous pull request before branching? (Im so confused abt git arrrr). Thx!

**Comment by dsirianni** [comment]
I'll try to address your questions, but we can talk more about it after class tomorrow or something in the future.

> (but I though I did. I just delete the branch when you accepted my week01 pull request)

Both of your PRs seem like they have originated from your master branch (see the "...merged 3 commits into `GT-IDEaS:master` from `andrewytp:master`..." line at the top of the page, under the title of your PR).  It's always best to keep your `master` branch up-to-date, and only checkout new branches _from_ this up-to-date master.  

> Or is that because I have to wait until you confirm my previous pull request before branching? 

Yes, that's exactly why your old commits were in the new PR (since both of your PRs seem to have originated from your `master`).  Any time a branch (let's call it `feature` has commits on top of the base (the point at which you created the branch) and you create a new branch `new-feature` from that one, the `new-feature` branch automatically incorporates those commits into it.  That's why it's always best to branch off of an up-to-date `master`.  You don't need to wait until a PR gets merged before you create a new branch -- just go back to `master`, update it, and checkout a new branch from there.  Since your existing PR should be within its own previous branch, you can work on your new branch independently of the old one.

> (Im so confused abt git arrrr)

Don't worry! Git/GitHub take some time to get used to.  It took me a lot longer to get to using branches effectively than I'm sure it will take you. Just keep practicing!

>  I am not sure the cause and how to avoid it next time.

Basically, just keep your `local/master` branch up-to-date, and always update it before creating a new branch.  That should be it!
